[pytorch] Composite backend potential fix for is_backend_available

wesbland opened the pull request:
Summary: `is_backend_available` takes in a string and expects it to only be backend, if its given a composite (device:backend) string, it fails.

Reviewed By: prashrock

Differential Revision: D81886736




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

kwen2501 reviewed (commented):
Are you proposing a UI like this?
```
is_backend_available("cuda:nccl")
```
May I know why a user would pass in `"cuda:nccl"` instead of just `is_backend_available("nccl")` ?

H-Huang reviewed (approved):
Seems reasonable to me

wesbland committed:
[pytorch]Composite backend potential fix for is_backend_available (#165061)

Summary:

`is_backend_available` takes in a string and expects it to only be backend, if its given a composite (device:backend) string, it fails.

Reviewed By: prashrock

Differential Revision: D81886736
